src/root.component.tsx

```diff
@@ -1,14 +1,46 @@
 import React from "react";
-import { Route } from "react-router";
-import { BrowserRouter } from "react-router-dom";
+import { BrowserRouter, Route } from "react-router-dom";
 import openmrsRootDecorator from "@openmrs/react-root-decorator";
 import PatientChartSummary from "./summary/patient-chart-summary.component";
+import PatientBanner from "./summary/banner/patient-banner.component";
+import { getCurrentPatient } from "@openmrs/esm-api";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import { LevelTwoRoutes } from "./summary/level-two-routes";
 
 function Root(props) {
+  const [currentPatient, setCurrentPatient] = React.useState(null);
+  const [showPatientSummary, setShowPatientSummary] = React.useState(true);
+
+  React.useEffect(() => {
+    const subscription = getCurrentPatient().subscribe(patient => {
```

Comment:
I don't see value in passing the current patient down as a prop from this component to other components. Each child component can just subscribe to the same observable to get the patient.

Doing this is setting us up for "prop plumbing" which is something that a lot of devs think is annoying (even though I think it's usually fine). I just don't really see value though in [lifting the state up](https://reactjs.org/docs/lifting-state-up.html#lifting-state-up) here.

Reply:
@joeldenning I see what you mean, can this be addressed in a separate PR as it will involve a lot of changes and refactor?